apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {ethers} from 'ethers';
+import useSWR from 'swr';
+import {useActionFlow} from '@vaults/contexts/useActionFlow';
+import {Solvers, useSolver} from '@vaults/contexts/useSolver';
+import {getEthZapperContract} from '@vaults/utils';
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useSettings} from '@yearn-finance/web-lib/contexts/useSettings';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {useChainID} from '@yearn-finance/web-lib/hooks/useChainID';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {ETH_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+import {defaultTxStatus} from '@yearn-finance/web-lib/utils/web3/transaction';
+import {useWallet} from '@common/contexts/useWallet';
+import {useAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import {DefaultTNormalizedBN} from '@common/utils';
+
+import type {ReactElement} from 'react';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import type {TNormalizedBN} from '@common/types/types';
+
+function	VaultDetailsQuickActionsButtons(): ReactElement {
+	const {refresh} = useWallet();
+	const {isActive} = useWeb3();
+	const {safeChainID} = useChainID();
+	const {networks} = useSettings();
+	const [txStatusApprove, set_txStatusApprove] = useState(defaultTxStatus);
+	const [txStatusDeposit, set_txStatusDeposit] = useState(defaultTxStatus);
+	const [allowanceFrom, set_allowanceFrom] = useState<TNormalizedBN>();
+	const {
+		selectedOptionFrom, selectedOptionTo,
+		amount, onChangeAmount,
+		maxDepositPossible, isDepositing
+	} = useActionFlow();
+	const {onApprove, onExecuteDeposit, onExecuteWithdraw, currentSolver} = useSolver();
+	const retrieveAllowance = useAllowanceFetcher();
+
+	const shouldUseVanillaAllowance = [Solvers.VANILLA, Solvers.CHAIN_COIN, Solvers.PARTNER_CONTRACT].includes(currentSolver);
```

Comment:
For example, move this to a 
```typescript
function shouldUseVanillaAllowance(currentSolver: Solvers): boolean {
	return [Solvers.VANILLA, Solvers.CHAIN_COIN, Solvers.PARTNER_CONTRACT].includes(currentSolver);
}
```
and have a comment on top because it's a bit unclear what is the "vanilla" allowance, wdyt?